R/convert_sim_var.R

```diff
@@ -0,0 +1,70 @@
+
+#' convert an existing simulation variable into a different one
+#' 
+#' allows you to provide specific variable or unit conversions to create a new variable 
+#' that can be used with \code{\link{plot_var_compare}}, and others. 
+#' 
+#' @param nc_file the netcdf output file
+#' @param \dots an expression to convert variable based on other variables or offsets
+#' @param unit the units for the new variable
+#' @param longname the longname for the new variable
+#' @export
+#' @importFrom lazyeval lazy_dots lazy_eval
+#' @importFrom ncdf4 ncvar_def ncvar_put ncvar_add
+#' @examples 
+#' \dontrun{
+#'sim_folder <- run_example_sim(verbose = FALSE)
+#'nc_file <- file.path(sim_folder, 'output.nc')
+#'convert_sim_var(nc_file, tempF = temp/5*9+32, unit='degF',longname='temperature degrees Farenheit')
+#'plot_var(nc_file, 'tempF')
+#'convert_sim_var(nc_file, crazy_var = temp-u_mean*1000)
+#'plot_var(nc_file, 'crazy_var')
+#'
+#' }
+#' 
+convert_sim_var <- function(nc_file='output.nc', ..., unit='', longname=''){
+  
+  sim.vars <- sim_vars(nc_file)$name
+  message('convert_sim_var is untested and in development')
+  
+  # // here, vals would be defined by the function passed in by `...`. Probably captured w/ lazyeval?
+  # lazyeval::lazy_eval(convert, data=list(temp=ncvar_get(nc, 'temp')))
+  convert <- lazyeval::lazy_dots(...)
+  if (length(convert) > 1)
+    stop('not yet ready to handle multi-var expressions')
+  
+  var.name <- names(convert)
+  if (var.name %in% sim.vars)
+    stop(var.name, ' cannot be added, it already exists.', call. = FALSE)
+  
+  fun.string <- deparse(convert[[1]]$expr)
+  variables <- strsplit(fun.string,"[^a-zA-Z_]")[[1]]
+  variables <- variables[variables != '']
+  vars.not.in <- variables[!variables %in% sim.vars]
```

Comment:
to support the function use that Luke mentioned above, would want to change this to `nc.vars`, and do
```r
data <- lapply(nc.vars, function(v) get_raw(nc_file, v))
```
Then let the function call fail if bad names are used.